lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -2033,14 +2033,9 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
   }
 
   /// Determine the number of bits set.
-  static unsigned numBitsSet(uint64_t value) {
-    unsigned count = 0;
-    for (uint64_t i : range(0, 63)) {
-      if (value & (uint64_t(1) << i))
-        ++count;
-    }
+  static inline unsigned numBitsSet(uint64_t value) {
 
-    return count;
+    return static_cast<unsigned>(__builtin_popcountll(value));
```

Comment:
I don't think that this is portable.  This code is definitely built with MSVC, so this would be a lot of complexity for very little gain.  It is also unclear if the premise is sound: loop idiom recognizer should recognize what is going on.